ioSPI/cryoemio.py

```diff
@@ -0,0 +1,362 @@
+"""Helper functions for tem.py processing of input and output files."""
+
+import random
+
+import h5py
+import mrcfile
+import numpy as np
+import yaml
+
+
+def mrc2data(mrc_file):
+    """Return micrograph from an input .mrc file.
+
+    Parameters
+    ----------
+    mrc_file : str
+        File name for .mrc file to turn into micrograph
+    """
+    with mrcfile.open(mrc_file, "r", permissive=True) as mrc:
+        micrograph = mrc.data
+    if len(micrograph.shape) == 2:
+        micrograph = micrograph[np.newaxis, ...]
+    return micrograph
+
+
+def data_and_dic2hdf5(data, h5_file):
+    """Convert dictionary data to hdf5 file format.
+
+    Parameters
+    ----------
+    data : dict
+        Dictionary of data to save.
+    h5_file : str
+        Relative path to h5 file.
+    """
+    dic = {"data": data}
+    with h5py.File(h5_file, "w") as file:
+        recursively_save_dict_contents_to_group(file, "/", dic)
+
+
+def recursively_save_dict_contents_to_group(h5file, path, dic):
+    """Recursively save dictionary contents to group.
+
+    Parameters
+    ----------
+    h5file : File
+        .hdf5 file to write to.
+    path : str
+        Relative path to save dictionary contents.
+    dic : dict
+        Dictionary containing data.
+    """
+    for k, v in dic.items():
+        if isinstance(v, (np.ndarray, np.int64, np.float64, int, float, str, bytes)):
+            h5file[path + k] = v
+        elif isinstance(v, type(None)):
+            h5file[path + k] = str("None")
+        elif isinstance(v, dict):
+            recursively_save_dict_contents_to_group(h5file, path + k + "/", v)
+        else:
+            raise ValueError("Cannot save %s type" % type(v))
+
+
+def fill_parameters_dictionary(
+    yaml_file, mrc_file, pdb_file, crd_file, log_file, dose=None, noise=None
+):
+    """Return parameter dictionary with settings for simulation.
+
+    Parameters
+    ----------
+    yaml_file : str
```

Comment:
eventually there will be many yaml files with different entries. a more informative name would be better.

Reply:
updated to input_params_file